### PR TITLE
feat: impl `zks_getAllAccountBalances` and `zks_getConfirmedTokens`

### PR DESCRIPTION
**Pull request by grw-ms**
# What :computer: 
* impl `zks_getAllAccountBalances`
* impl `zks_getConfirmedTokens`

# Why :hand:
* Compatibility with mainnet RPC

# Evidence :camera:
Screenshot 2023-10-20 at 14 44 36
Screenshot 2023-10-20 at 14 40 02
Screenshot 2023-10-20 at 14 39 50

# Notes :memo:
* Fixes https://github.com/matter-labs/era-test-node/issues/142

### CODE REVIEW

File: e2e-tests/test/zks-apis.test.ts [hunk 1]
```diff
@@ -152,3 +152,18 @@ describe("zks_getRawBlockTransactions", function () {
     expect(txns[0]["execute"]["calldata"]).to.equal(receipt.data);
   });
 });
+
+describe("zks_getConfirmedTokens", function () {
+  it("Should return empty list", async function () {
+    const tokens = await provider.send("zks_getConfirmedTokens", [0, 100]);
+    expect(tokens.length).to.equal(0);
```
Comment (MexicanAce) [inline e2e-tests/test/zks-apis.test.ts:159]: I think we should by default at least include the `ETH` token, as I think a non-empty response is necessary for the Block Explorer/Portal : 

```json
{
"l1Address":"0x0000000000000000000000000000000000000000",
"l2Address":"0x0000000000000000000000000000000000000000",
"name":"Ether",
"symbol":"ETH",
"decimals":18
}
```

File: src/zks.rs [hunk 2]
```diff
@@ -241,13 +265,49 @@ impl<S: Send + Sync + 'static + ForkSource + std::fmt::Debug> ZksNamespaceT
...
+        let inner = self.node.clone();
+        Box::pin({
+            let address = address.clone();
+            self.get_confirmed_tokens(0, 100)
+                .then(move |tokens| async move {
+                    let tokens =
+                        tokens.map_err(|_err| into_jsrpc_error(Web3Error::InternalError))?;
+
+                    let mut writer = inner
+                        .write()
```
Comment (MexicanAce) [inline src/zks.rs:292]: Could this be a `.read()`? I don't think we're updating any values within this API.

File: test_endpoints.http [hunk 3]
```diff
@@ -729,6 +729,28 @@ content-type: application/json
...
+
+###
+POST http://localhost:8011
+content-type: application/json
+
+{
+    "jsonrpc": "2.0",
+    "id": "1",
+    "method": "zks_getAllAccountBalances",
+    "params": ["0x364d6D0333432C3Ac016Ca832fb8594A8cE43Ca6"]
```
Comment (MexicanAce) [inline test_endpoints.http:747]: nit: Could we update this to be one of the Rich Wallet addresses? Just so we know it'll return something a balance `era_test_node run`

File: e2e-tests/test/zks-apis.test.ts (hunk 1 above)
Comment (grw-ms) [reply e2e-tests/test/zks-apis.test.ts]: done

File: src/zks.rs (hunk 2 above)
Comment (grw-ms) [reply src/zks.rs:297]: i don't think so- we may mutate the cache- `fork_storage.read_value()` takes `&mut self`

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**